packages/manager/apps/catalog-revamp/src/e2e/FilterAndSearchBehavior.test.ts

```diff
@@ -0,0 +1,75 @@
+import { test, expect } from '@playwright/test';
+import '@playwright-helpers/login';
+
+test('should filter results based on Bare Metal Cloud universe', async ({
+  page,
+}) => {
+  const buttons = await page.$$('osds-button');
+  const buttonFilter = buttons[0];
+  if (buttonFilter) await buttonFilter.click();
+  await page.waitForTimeout(1000);
+
+  const checkboxes = await page.$$('osds-checkbox-button');
+  const checkboxeBarMetal = checkboxes[0];
+  if (checkboxeBarMetal) await checkboxeBarMetal.click();
+
+  const applyFiltersButton = buttons[1];
+  if (applyFiltersButton) await applyFiltersButton.click();
+
+  await page.waitForTimeout(1000);
+
+  const products = await page.$$('msc-tile');
+  expect(products.length).toBe(5);
+});
+
+test('should clear all filters when Clear All button is clicked', async ({
+  page,
+}) => {
+  const buttons = await page.$$('osds-button');
+  const buttonFilter = buttons[0];
+  if (buttonFilter) await buttonFilter.click();
+
+  const checkboxes = await page.$$('osds-checkbox-button');
+  const checkboxeBarMetal = checkboxes[0];
+  if (checkboxeBarMetal) await checkboxeBarMetal.click();
+
+  const applyFiltersButton = buttons[1];
+  if (applyFiltersButton) await applyFiltersButton.click();
+
+  await page.waitForTimeout(1000);
+
+  let products = await page.$$('msc-tile');
+  expect(products.length).toBe(5); // 5 products on bar metal cloud
+
+  if (buttonFilter) await buttonFilter.click();
+  const links = await page.$$('osds-link');
+  const resetFilterLink = links[0];
+  if (resetFilterLink) await resetFilterLink.click();
+
+  await page.waitForTimeout(1000);
+  products = await page.$$('msc-tile');
+  expect(products.length).toBe(40); // All products
+});
+
+// Scenario: No results after filtering
+test('should show "No results found" when filters match no products', async ({
+  page,
+}) => {
+  const buttons = await page.$$('osds-button');
+  const buttonFilter = buttons[0];
+  if (buttonFilter) await buttonFilter.click();
+
+  const checkboxes = await page.$$('osds-checkbox-button');
+  const checkboxeBarMetal = checkboxes[0];
+  if (checkboxeBarMetal) await checkboxeBarMetal.click();
+  const checkboxeAIMachineLearning = checkboxes[6];
+  if (checkboxeAIMachineLearning) await checkboxeAIMachineLearning.click();
+
+  const applyFiltersButton = buttons[1];
+  if (applyFiltersButton) await applyFiltersButton.click();
+
+  await page.waitForTimeout(1000);
+
+  const products = await page.$$('msc-tile');
+  expect(products.length).toBe(0); // 0 products : no result
```

Comment:
Can you test if the message is displayed pls

Reply:
Done